docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -0,0 +1,1382 @@
+
+DWARF debug info is generated in a separate file and can even be omitted in the default deployment and only be transferred and used on demand,
+for profiling or debugging purposes.
+Furthermore, the presence of debug info enables `perf` to show us the relevant source code lines as well,
+hence it does not bloat the native binary itself.
+To do that, simply call `perf report` with an extra parameter to show source code lines:
+
+[source,bash]
+----
+perf report --stdio -F+srcline
+...
+83.69%     0.00%  GreetingResource.java:20 ...
+...
+83.69%     0.00%  AbstractStringBuilder.java:1025 ...
+...
+83.69%     0.00%  ArraycopySnippets.java:95 ...
+----
+
+The performance penalty of preserving the frame pointer is due to using the extra register for stack walking,
+particularly in `x86_64` compared to `aarch64` where there are less registers available.
+Using this extra register reduces the number of registers that are available for other work,
+which can lead to performance penalties.
+
+=== I think I’ve found a bug in native-image, how can I debug it with the IDE?
+
+Although it is possible to remote debug processes within containers,
+it might be easier to step-by-step debug native-image by installing Mandrel locally and adding it to the path of the shell process.
+
+Native executable generation is the result of two Java processes that are executed sequentially.
+The first process is very short and it’s main job is to set things up for the second process.
+The second process is the one that takes care of most of the work.
+The steps to debug one process or the other vary slightly.
+
+Let’s discuss first how to debug the second process,
+which is the one you most likely to want to debug.
+The starting point for the second process is the `com.oracle.svm.hosted.NativeImageGeneratorRunner` class.
+To debug this process, simply add `--debug-attach=*:8000` as an additional build time argument:
+
+[source,bash,subs=attributes+]
+----
+./mvnw package -DskipTests -Pnative \
+    -Dquarkus.native.additional-build-args=--debug-attach=*:8000
+----
+
+The starting point for the first process is the `com.oracle.svm.driver.NativeImages` class.
+In GraalVM CE distributions, this first process is a binary, so debugging it in the traditional way with a Java IDE is not possible.
+However, Mandrel distributions (or locally built GraalVM CE instances) keep this as a normal Java process,
+so you can remote debug this process by adding the `--vm.agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:8000` as an additional build argument, e.g.
+
+[source,bash,subs=attributes+]
+----
+$ ./mvnw package -DskipTests -Pnative \
+    -Dquarkus.native.additional-build-args=--vm.agentlib:jdwp=transport=dt_socket\\,server=y\\,suspend=y\\,address=*:8000
+----
+
+=== Can I use JFR/JMC to debug to profile native binaries?
+
+JFR and JMC can be used to debug profile native binaries since GraalVM CE 21.2.
```

Comment:
As JFR and JMC have not been mentioned before, best at least have a link to them.